run.go

```diff
@@ -57,6 +57,10 @@ command(s) that get executed on start, edit the args parameter of the spec. See
 			Name:  "no-new-keyring",
 			Usage: "do not create a new session keyring for the container.  This will cause the container to inherit the calling processes session key",
 		},
+		cli.IntFlag{
+			Name:  "preserve-fds",
+			Usage: "Pass N additional file descriptors to the container (stdio + $LISTEN_FDS + N in total)",
```

Comment:
The current interface doesn't allow you to pass through fds 0, 1, or 2 in this way so I think [the SQLlite reasoning](https://www.sqlite.org/compile.html#minimum_file_descriptor) for avoiding those fds doesn't really apply here.

I can take a look switching things to behave like you suggest but it would have to be contingent on Go's [`os/exec`](https://golang.org/pkg/os/exec/) `Cmd.ExtraFiles` supporting the necessary semantics (i.e. a bug slice full of `nil`s apart from at offset `512` and `1337` in your example). Personally I don't think it is terribly useful to be able to pass arbitrary fds, it's pretty easy (and arguably more controlled) for the caller to `dup2` things into the right place after `fork` but before `exec` of runc.